yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -1337,55 +1337,8 @@ object Client extends Logging {
       conf: Configuration,
       credentials: Credentials) {
     if (shouldGetTokens(sparkConf, "hive") && UserGroupInformation.isSecurityEnabled) {
```

Comment:
Actually I don't really think the delegation tokens are needed in client mode (for Hive only). Only the driver talks to the metastore.